3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/WeatherEnrichment.scala

```diff
@@ -0,0 +1,186 @@
+import com.snowplowanalytics.weather.providers.openweather.Responses._
+
+// Iglu
+import iglu.client.SchemaCriterion
+
+// This project
+import utils.ScalazJson4sUtils
+import enrichments.EventEnrichments
+
+/**
+ * Companion object. Lets us create an WeatherEnrichment instance from a JValue
+ */
+object WeatherEnrichmentConfig extends ParseableEnrichment {
+
+  val supportedSchema = SchemaCriterion("com.snowplowanalytics.snowplow.enrichments", "weather_enrichment_config", "jsonschema", 1, 0)
+
+  def parse(config: JValue, schemaKey: SchemaKey): ValidatedNelMessage[WeatherEnrichment] = {
+    isParseable(config, schemaKey).flatMap { conf => {
+      (for {
+        apiKey       <- ScalazJson4sUtils.extract[String](config, "parameters", "apiKey")
+        cacheSize    <- ScalazJson4sUtils.extract[Int](config, "parameters", "cacheSize")
+        geoPrecision <- ScalazJson4sUtils.extract[Int](config, "parameters", "geoPrecision")
+        apiHost      <- ScalazJson4sUtils.extract[String](config, "parameters", "apiHost")
+        timeout      <- ScalazJson4sUtils.extract[Int](config, "parameters", "timeout")
+        enrich       =  WeatherEnrichment(apiKey, cacheSize, geoPrecision, apiHost, timeout)
+      } yield enrich).toValidationNel
+    }}
+  }
+}
+
+
+/**
+ * Contains weather enrichments based on geo coordinates and time
+ *
+ * @param apiKey weather provider API KEY
+ * @param cacheSize amount of days with prefetched weather
+ * @param geoPrecision rounder for geo lat/long floating, which allows to use
+ *                     more spatial precise weather stamps
+ * @param apiHost address of weather provider's API host
+ * @param timeout timeout in seconds to fetch weather from server
+ */
+case class WeatherEnrichment(apiKey: String, cacheSize: Int, geoPrecision: Int, apiHost: String, timeout: Int) extends Enrichment {
+
+  val version = new DefaultArtifactVersion("0.1.0")
+
+  private lazy val client = OwmCacheClient(apiKey, cacheSize, geoPrecision, apiHost, timeout)
+
+  private val schemaUri = "iglu:org.openweathermap/weather/jsonschema/1-0-0"
+
+  private implicit val formats = DefaultFormats
+
+  /**
+   * Get weather context as JSON for specific event
+   * Any non-fatal error will return failure and thus whole event will be
+   * filtered out in future
+   *
+   * @param latitude enriched event optional latitude (probably null)
+   * @param longitude enriched event optional longitude (probably null)
+   * @param time enriched event optional time (probably null)
+   * @return weather stamp as self-describing JSON object
+   */
+  // It accepts Java Float (JFloat) instead of Scala's because it will throw NullPointerException
+  // on conversion step if `EnrichedEvent` has nulls as geo_latitude or geo_longitude
+  def getWeatherContext(latitude: Option[JFloat], longitude: Option[JFloat], time: Option[DateTime]): Validation[String, JObject] =
+    if (latitude == null || longitude == null || time == null) {
```

Comment:
I added this check because ran into NullPointerException on `Float => JFloat` conversion several times.
```scala
import java.lang.{ Float => JFloat }
class A {
  @BeanProperty var latitude: JFloat = _
}
val a = new A
def get(foo: Option[Float]) = 1  // It will rail NPE
get(Option(a.latitude))
```
I don't think there's a big chance above values could be nulls because we control calling code, but overall this check adding safety for me.